src/PHPCR/Util/ValueConverter.php

```diff
@@ -130,11 +130,6 @@ public function convertType($value, $type, $srctype = PropertyType::UNDEFINED)
             && $value instanceof NodeInterface
         ) {
             /** @var $value NodeInterface */
-            // In Jackrabbit a new node cannot be referenced until it has been persisted
-            // See: https://issues.apache.org/jira/browse/JCR-1614
-            if ($value->isNew()) {
-                throw new ValueFormatException('Node ' . $value->getPath() . ' must be persisted before being referenceable');
```

Comment:
There are 2 places, but it looks like you would only need to handle one (the other is explicitly a binary): https://github.com/jackalope/jackalope-jackrabbit/search?utf8=%E2%9C%93&q=convertType

Don't know what the best method would be. Maybe introduce a `convertType` method to the `Client` and just do:

````php
private function convertType($value, $type)
{
    if ($type === PropertyType::REFERENCE (...)) {
        // validate
    }
   
    return $this->valueConverter->convertValue($value, $type);
}
````

?